unified_planning/engines/factory.py

```diff
@@ -57,35 +69,126 @@ def format_table(header: List[str], rows: List[List[str]]) -> str:
     return '\n'.join(rows_str)
 
 
+def get_possible_config_locations() -> List[str]:
+    """Returns all the possible location of the configuration file."""
+    home = os.path.expanduser('~')
+    files = []
+    stack = inspect.stack()
+    p = os.path.dirname(os.path.abspath(stack[-1].filename))
+    while os.path.join(p, 'up.ini') not in files:
+        files.append(os.path.join(p, 'up.ini'))
+        p = os.path.abspath(os.path.join(p, os.pardir))
+    files.append(os.path.join(home, '.uprc'))
+    files.append(os.path.join(home, 'up.ini'))
+    return files
+
+
+def configure_factory(factory: 'Factory'):
+    """
+    Reads a configuration file and configures the factory.
+
+    The following is an example of configuration file:
+
+
+    [global]
+    engine_preference_list: fast-downward fast-downward-opt enhsp enhsp-opt tamer
+
+    [engine <engine-name>]
+    module_name: <module-name>
+    class_name: <class-name>
+
+
+    The configuration is read from the first up.ini file located in any of the
+    parent directories from which this code was called or, otherwise, from one
+    of the following files: ~/.uprc, ~/up.ini.
+    """
+    config = configparser.ConfigParser()
+    files = get_possible_config_locations()
+    config.read(files)
+
+    new_engine_sections = [s for s in config.sections()
+                           if s.lower().startswith("engine ")]
+
+    for s in new_engine_sections:
+        name = s[len("engine "):]
+
+        module_name = config.get(s, "module_name")
+        assert module_name is not None, ("Missing 'module_name' value in definition"
+                                         "of '%s' engine" % name)
+
+        class_name = config.get(s, "class_name")
+        assert class_name is not None, ("Missing 'class_name' value in definition"
+                                        "of '%s' engine" % name)
+
+        factory.add_engine(name, module_name, class_name)
+
+
+    if "global" in config.sections():
+        pref_list = config.get("global", "engine_preference_list")
+
+        if pref_list is not None:
+            prefs = pref_list.split()
```

Comment:
`split` can be dangerous if, for example I use two spaces in a row. my suggestion is to `trim` the resulting strings and skip empty strings

```suggestion
            prefs = [x.trim() for x in pref_list.split() if len(x.trim()) > 0] #or something like this
```